admin/tracking/class-tracking.php

```diff
@@ -54,14 +54,54 @@ public function __construct( $endpoint, $threshold ) {
 	 * Registers all hooks to WordPress.
 	 */
 	public function register_hooks() {
+		// Send tracking data on `admin_init`.
 		add_action( 'admin_init', array( $this, 'send' ), 1 );
+
+		// Add an action hook that will be triggered at the specified time by `wp_schedule_single_event()`.
+		add_action( 'wpseo_send_tracking_data_after_core_update', array( $this, 'send' ) );
+		// Call `wp_schedule_single_event()` after a WordPress core update.
+		add_action( 'upgrader_process_complete', array( $this, 'schedule_tracking_data_sending' ), 10, 2 );
+	}
+
+	/**
+	 * Schedules a new sending of the tracking data after a WordPress core update.
+	 *
+	 * @param bool|WP_Upgrader $upgrader Optional. WP_Upgrader instance or false.
+	 *                                   Depending on context, it might be a Theme_Upgrader,
+	 *                                   Plugin_Upgrader, Core_Upgrade, or Language_Pack_Upgrader.
+	 *                                   instance. Default false.
+	 * @param array            $data     Array of update data.
+	 *
+	 * @return void
+	 */
+	public function schedule_tracking_data_sending( $upgrader = false, $data = array() ) {
+		// Return if it's not a WordPress core update.
+		if ( ! $upgrader || ! isset( $data['type'] ) || $data['type'] !== 'core' ) {
+			return;
+		}
+
+		/*
+		 * To uniquely identify the scheduled cron event, `wp_next_scheduled()`
+		 * needs to receive the same arguments as those used when originally
+		 * scheduling the event otherwise it will always return false.
+		 */
+		if ( ! wp_next_scheduled( 'wpseo_send_tracking_data_after_core_update', true ) ) {
+			/*
+			 * Schedule sending of data tracking 6 hours after a WordPress core
+			 * update. Pass a `true` parameter for the callback `$force` argument.
+			 */
+			wp_schedule_single_event( ( time() + ( HOUR_IN_SECONDS * 6 ) ), 'wpseo_send_tracking_data_after_core_update', true );
```

Comment:
Awesome. For the time I had `strtotime( '+6hours' )` in mind. But this is also good 👍